File: packages/sdk-php/src/AccountAPI.php
<?php

declare(strict_types=1);

namespace Melaya;

/**
 * Account API — authenticated reads about your Melaya account.
 *
 * Maps to https://api.melaya.org/api/v1/private/*.
 * Connected-exchange key references (masked), tier limits, and live usage counters.
 */
class AccountAPI
{
    public function __construct(private readonly HttpClient $http) {}

    /**
     * The exchange API keys connected to your account.
     * `apiKey` is masked (display-only); use `apiKeyId` (e.g. `BINANCEUSDM_0`)
     * as the reference when launching strategies or minting a private stream ticket.
     */
    public function keys(): array
    {
        return $this->http->get('/api/v1/private/keys')['keys'];
    }

    /** Tier, plan limits, and live usage counters (mirrors the dashboard usage page). */
    public function usage(): array
    {
        return $this->http->get('/api/v1/private/usage');
    }

    /** Status of your platform API key (tier, max concurrent connections). */
    public function apiKeyStatus(): array
    {
        return $this->http->get('/api/v1/private/api-key');
    }
}

File: packages/sdk-java/src/main/java/org/melaya/AccountAPI.java
package org.melaya;

import com.fasterxml.jackson.databind.JsonNode;

/**
 * Account API — authenticated reads about your Melaya account.
 * Maps to {@code https://api.melaya.org/api/v1/private/*}.
 */
public class AccountAPI {

    private final HttpClient http;

    AccountAPI(HttpClient http) {
        this.http = http;
    }

    /**
     * The exchange API keys connected to your account.
     * {@code apiKey} is masked (display-only); use {@code apiKeyId} when launching strategies.
     */
    public JsonNode keys() {
        return http.get("/api/v1/private/keys", null).get("keys");
    }

    /** Tier, plan limits, and live usage counters (mirrors the dashboard's usage page). */
    public JsonNode usage() {
        return http.get("/api/v1/private/usage", null);
    }

    /** Status of your platform API key (tier, max concurrent connections). */
    public JsonNode apiKeyStatus() {
        return http.get("/api/v1/private/api-key", null);
    }
}
